usaspending_api/search/delta_models/subaward_search.py

```diff
@@ -0,0 +1,555 @@
+        fpds.product_or_service_code,
+        psc.description AS product_or_service_description,
+
+        COALESCE(UPPER(bs.sub_legal_entity_country_code), 'USA') AS sub_legal_entity_country_code,
+        rcc.country_name AS sub_legal_entity_country_name,
+        LPAD(CAST(CAST(REGEXP_EXTRACT(rec.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0') AS sub_legal_entity_county_code,
+        rec.county_name AS sub_legal_entity_county_name,
+        LEFT(COALESCE(bs.sub_legal_entity_zip, ''), 5) AS sub_legal_entity_zip5,
+        rec.census_code AS sub_legal_entity_city_code,
+        LPAD(CAST(CAST(REGEXP_EXTRACT(UPPER(bs.sub_legal_entity_congressional), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0') AS sub_legal_entity_congressional,
+
+        fabs.place_of_performance_scope AS place_of_perform_scope,
+        COALESCE(UPPER(bs.sub_place_of_perform_country_co), 'USA') AS sub_place_of_perform_country_co,
+        pcc.country_name AS sub_place_of_perform_country_name,
+        LPAD(CAST(CAST(REGEXP_EXTRACT(pop.county_numeric, '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 3, '0') AS sub_place_of_perform_county_code,
+        pop.county_name AS sub_place_of_perform_county_name,
+        LEFT(COALESCE(sub_place_of_performance_zip, ''), 5) AS sub_place_of_perform_zip5,
+        pop.census_code AS sub_place_of_perform_city_code,
+        LPAD(CAST(CAST(REGEXP_EXTRACT(UPPER(bs.sub_place_of_perform_congressio), '^[A-Z]*(\\d+)(?:\\.\\d+)?$', 1) AS SHORT) AS STRING), 2, '0') AS sub_place_of_perform_congressio
+
+    FROM
+        raw.broker_subaward AS bs
+    LEFT OUTER JOIN
+        raw.awards AS a
+            ON a.generated_unique_award_id = bs.unique_award_key
+    LEFT OUTER JOIN
+        raw.transaction_normalized AS tn
+            ON tn.id = a.latest_transaction_id
+    LEFT OUTER JOIN
+        raw.transaction_fpds AS fpds
+            ON fpds.transaction_id = a.latest_transaction_id
+    LEFT OUTER JOIN
+        raw.transaction_fabs AS fabs
+            ON fabs.transaction_id = a.latest_transaction_id
+    LEFT OUTER JOIN
+        global_temp.agency AS aa
+            ON aa.id = a.awarding_agency_id
+    LEFT OUTER JOIN
+        global_temp.toptier_agency AS taa
+            ON taa.toptier_agency_id = aa.toptier_agency_id
+    LEFT OUTER JOIN
+        global_temp.subtier_agency AS saa
+            ON saa.subtier_agency_id = aa.subtier_agency_id
+    LEFT OUTER JOIN
+        global_temp.agency AS fa
+            ON fa.id = a.funding_agency_id
+    LEFT OUTER JOIN
+        global_temp.toptier_agency AS tfa
+            ON tfa.toptier_agency_id = fa.toptier_agency_id
+    LEFT OUTER JOIN
+        global_temp.subtier_agency AS sfa
+            ON sfa.subtier_agency_id = fa.subtier_agency_id
+    LEFT OUTER JOIN
+        (
+            SELECT
+                faba.award_id,
+                SORT_ARRAY(COLLECT_SET(DISTINCT taa.treasury_account_identifier::INTEGER)) AS treasury_account_identifiers
```

Comment:
The DISTINCT is redundant since COLLECT_SET will only collect unique items for the set.

Reply:
Does `taa.treasury_account_identifier::INTEGER` need to be changed to CAST(taa.treasury_account_identifier AS INTEGER)?

Reply:
> Does taa.treasury_account_identifier::INTEGER need to be changed to CAST(taa.treasury_account_identifier AS INTEGER)?

It does! I think I have that updated in the following branch. I'll copy it here.